twenty-first/src/util_types/algebraic_hasher.rs

```diff
@@ -55,9 +55,17 @@ pub trait SpongeHasher: Clone + Debug + Default + Send + Sync {
                 .collect::<Vec<_>>()
                 .try_into()
                 .expect("a multiple of RATE elements");
-            Self::absorb(sponge, &absorb_elems);
+            Self::absorb_once(sponge, &absorb_elems);
         }
     }
+
+    fn pad_and_absorb_repeatedly(sponge: &mut Self::SpongeState, input: &[BFieldElement]) {
```

Comment:
I prefer `pad_and_absorb_all` over `pad_and_absorb_repeatedly`. The reason is that "repeatedly" is confusing when only one iteration is necessary.

Sorry for the unstable recommendation.

Reply:
Should the name `absorb_repeatedly` also be change to `absorb_all` then?

Reply:
Nah, "repeatedly" is fine if it is the opposite of "once". For `pad_and_absorb_all` there is no "once" case.